gdlsc.go

```diff
@@ -1,104 +1,94 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
 	"os"
-	"os/exec"
+	"path"
+	"path/filepath"
 	"strings"
+
+	"github.com/ryanuber/go-license"
 )
 
-// URL represents path a github repos license
-var URL = "https://api.github.com/repos/%s/license?access_token=" + os.Getenv("ACCESS_TOKEN")
+const root = "vendor/"
 
-//types for message to be printed
-type message struct {
-	Dependency string
-	License    string
-	Link       string
-}
+var files []string
+var rootPath string
+var noLicense = map[string]bool{}
 
-// options for response from github
-type response struct {
-	Html_URL string
-	License  license
+// Response contains license and file info
+type Response struct {
+	License string
+	File    string
 }
 
-type license struct {
-	Name string
+func getLicense(input chan string, output chan Response) {
+	for file := range input {
+		l, err := license.NewFromFile(file)
+		splitFile := strings.Split(file, "/")
+		splitFile = splitFile[1 : len(splitFile)-1]
+		file := path.Join(splitFile...)
+		if err != nil {
+			output <- Response{File: file, License: err.Error()}
+		} else {
+			output <- Response{File: file, License: l.Type}
+		}
+	}
+}
+func checkLicense(path string) bool {
+	path = strings.ToUpper(path)
+	return strings.Contains(path, "LICENSE") || strings.Contains(path, "COPYING")
 }
 
-// uses github api to fetch license based on link generated
-func getLicense(dependency string, ch chan message) {
-	url := formatLink(dependency)
-	resp, err := http.Get(url)
+func walker(filePath string, f os.FileInfo, err error) error {
 	if err != nil {
-		panic(err)
+		fmt.Println("No vendor folder found. Aborting..")
+		return err
 	}
 
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		panic(err)
+	if !f.IsDir() {
```

Comment:
I think a slightly more correct version of this should be if f.IsRegular(). This is also not a big deal, since it's very unlikely, but just saying it FYI. On unix systems there are a lot of weird type of files like sockets, devices, symlinks, etc. What you are looking for here is plain data files.